### PR TITLE
Remove localization workaround in Data Annotation Attributes

### PR DESCRIPTION
**Pull request by hishamco**
Two years ago when I started [Support data annotations PO localization](https://github.com/OrchardCMS/OrchardCore/pull/4675) PR, there are long discussion to support localization in data annotations, the PR takes so long to merge ;)

Back in the day @agriffard did workaround to achive the goal by using `IValidatableObject`, now it's the time to remove this workaround and use of the new way of doing this

### CODE REVIEW

**Comment by hishamco** [comment]
FYI I added `EmailAddressAttribute` to validate emails using our `IEmailAddressValidator`

**Comment by hishamco** [comment]
Strange!! why the build is not completed yet?!!

**Comment by Skrypt** [comment]
How does the poextractor tool works with these?

**Comment by hishamco** [comment]
We already added a mechanism to localize the data annotations default messages, in case if you want supply a custom message you need to add them to `{culture}.po`

Let me check if the poextractor tool already handle this or not

/cc @lukaskabrt

**Comment by hishamco** [comment]
Seems it's not there, I need to implement this on a tool

**Comment by hishamco** [comment]
Please hold this until support data annotations in PoExtractor tool

**Comment by hishamco** [comment]
@Skrypt do you have any other comment or shall we approve it

File: src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Email/EmailAddressAttribute.cs
Comment (hishamco) [reply src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Email/EmailAddressAttribute.cs:18]: Nope, this similar to normal `EmailAddressAttribute` behavior, so we don't wanna confuse others

Comment (Skrypt) [reply src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Email/EmailAddressAttribute.cs:18]: Here the method returns a ModelState to the controller so if I would pass a NULL email it would let it pass as a valid email. The only reason this line will never get hit is that we have this condition in the controller :  

```C#
            if (String.IsNullOrEmpty(model.Email))
            {
                ModelState.AddModelError("Email", S["Email is required."]);
            }
``` 

So, if the intent is to return if the email IsValid here then it fails to do so. Because a NULL email is not an email. So the ModelState returned here I think is wrong.

Comment (Skrypt) [reply src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Email/EmailAddressAttribute.cs:18]: I will let others review this one but I think it's wrong unless it is by design.

Comment (Skrypt) [reply src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Email/EmailAddressAttribute.cs:18]: You mean this : 

https://github.com/microsoft/referencesource/blob/5697c29004a34d80acdaf5742d7e699022c64ecd/System.ComponentModel.DataAnnotations/DataAnnotations/EmailAddressAttribute.cs#L23

Then I agree

Comment (hishamco) [reply src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Email/EmailAddressAttribute.cs:18]: Exactly, that's why I did the condition explicitly in the control. I know it seems weird but it's by design

File: src/OrchardCore/OrchardCore.Application.Cms.Core.Targets/OrchardCore.Application.Cms.Core.Targets.targets
Comment (Skrypt) [reply src/OrchardCore/OrchardCore.Application.Cms.Core.Targets/OrchardCore.Application.Cms.Core.Targets.targets:30]: Revert

Comment (hishamco) [reply src/OrchardCore/OrchardCore.Application.Cms.Core.Targets/OrchardCore.Application.Cms.Core.Targets.targets:30]: Oops, seems I push it this acciendetally. Jasim was right I did that to avoid building po files everytime and ignore my changes

**Comment by hishamco** [comment]
@agriffard your approve if you don't have any doubt

**Comment by hishamco** [comment]
@sebastienros can this ship in `1.4.0`?

**Review by sebastienros — APPROVED** [review]

**Comment by sebastienros** [comment]
I like it, even if the interface based solution is also very flexible in terms of custom code.

**Pull request merged**
